db/document.go

```diff
@@ -167,6 +168,14 @@ func (sd *SyncData) SetCV(hlv *HybridLogicalVector) {
 	sd.RevAndVersion.CurrentVersion = string(base.Uint64CASToLittleEndianHex(hlv.Version))
 }
 
+func (sd *SyncData) hasFlag(flag uint8) bool {
+	return sd.Flags&flag != 0
+}
+
+func (sd *SyncData) IsDeleted() bool {
```

Comment:
I think get in the habit of writing docstrings, especially for public functions.


```suggestion
// IsDeleted returns true if the document metadata indicates it is a tombstone.
func (sd *SyncData) IsDeleted() bool {
```